### PR TITLE
Postgres 10 Upgrade

### PR DESCRIPTION
**Pull request by rooftopcellist**
##### SUMMARY
This PR upgrades Postgres from 9.6 --> 10.  

Related Issue: https://github.com/ansible/awx/issues/3778 Upgrade to PostgreSQL 10 or 11

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

 - API
 - Installer

### CODE REVIEW

File: requirements/requirements.in [hunk 1]
```diff
@@ -29,7 +29,7 @@ Markdown==2.6.11    # used for formatting API help
 ordereddict==1.1
 prometheus_client==0.6.0
 psutil==5.4.3
-psycopg2==2.7.3.2   # problems with Segmentation faults / wheels on upgrade
+psycopg2==2.8.2
```
Comment (AlanCoding) [inline requirements/requirements.in:33]: nice comment

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -155,7 +155,7 @@ If you start a second terminal session, you can take a look at the running conta
 $ docker ps
 CONTAINER ID        IMAGE                                              COMMAND                  CREATED             STATUS              PORTS                                                                                                                                      NAMES
 aa4a75d6d77b        gcr.io/ansible-tower-engineering/awx_devel:devel   "/tini -- /bin/sh ..."   23 seconds ago      Up 15 seconds       0.0.0.0:5555->5555/tcp, 0.0.0.0:6899-6999->6899-6999/tcp, 0.0.0.0:8013->8013/tcp, 0.0.0.0:8043->8043/tcp, 22/tcp, 0.0.0.0:8080->8080/tcp   tools_awx_1
-e4c0afeb548c        postgres:9.6                                       "docker-entrypoint..."   26 seconds ago      Up 23 seconds       5432/tcp                                                                                                                                   tools_postgres_1
+e4c0afeb548c        postgres:10.6                                       "docker-entrypoint..."   26 seconds ago      Up 23 seconds       5432/tcp                                                                                                                                   tools_postgres_1
```
Comment (ryanpetrello) [inline CONTRIBUTING.md]: 👀

File: requirements/requirements.in (hunk 1 above)
Comment (rooftopcellist) [reply requirements/requirements.in:33]: For reference: https://github.com/psycopg/psycopg2/issues/543

Comment (AlanCoding) [reply requirements/requirements.in:33]: wow that issue has a whole lot of backlinks. It's closed, so do you have a sense of their resolution? At a quick look, it sounds like they recommended the package `psycopg2-binary` to do a wheel install. Could we switch to that and avoid the pin?

Comment (AlanCoding) [reply requirements/requirements.in:33]: https://pypi.org/project/psycopg2-binary/

ok, no

> The binary package is a practical choice for development and testing but in production it is advised to use the package built from sources.

I'm kind of unclear about where that leaves users of this library who want it in production.

Comment (rooftopcellist) [reply requirements/requirements.in:33]: At it's core, this is an issue between conflicting versions of libssl, and should manifest when you import psycopg2.  

Dev Container (CentOS 7): 
image

I'll check to make sure this doesn't segfault on RHEL7 & RHEL8 too.

Comment (AlanCoding) [reply requirements/requirements.in:33]: oh yeah, I wasn't following before, but the upgrade here looks good 👍

**Comment by piroux** [comment]
Hi !
For postgres, the pre-10 series released versions were following the `X.y.z` where `X.y` were the major part version and `z` the minor part.
Now with the post-10 released versions, they are following a new pattern `X.y` where `X` is the major and `y` the minor part.

According to their terminology:
> We always recommend that all users run the latest available minor release for whatever major version is in use.
> 
> Major versions usually change the internal format of system tables and data files. These changes are often complex, so we do not maintain backward compatibility of all stored data. A dump/reload of the database or use of the pg_upgrade module is required for major upgrades. We also recommend reading the upgrading section of the major version you are planning to upgrade to.
> 
> Upgrading to a minor release does not normally require a dump and restore; you can stop the database server, install the updated binaries, and restart the server. For some releases, manual changes may be required to complete the upgrade, so always read the release notes before upgrading.
> 
> While upgrading will always contain some level of risk, PostgreSQL minor releases fix only frequently-encountered bugs, security issues, and data corruption problems to reduce the risk associated with upgrading. For minor releases, the community considers not upgrading to be riskier than upgrading.

https://www.postgresql.org/support/versioning/

All this to say: if the previously chosen tag was `9.6`, only major without minor, I think we might as well continue to only set the major version, so that it would be `10` now and not `10.6`.

**Comment by rooftopcellist** [comment]
@piroux We have chosen to pin to the minor release of PostgreSQL 10.6 to maintain a consistent minor version across all of our supported platforms, as some platforms are currently locked in at 10.6.

**Comment by rooftopcellist** [comment]
recheck

File: CONTRIBUTING.md (hunk 2 above)
Comment (wenottingham) [reply CONTRIBUTING.md]: this should just be `postgres:10`

Comment (wenottingham) [reply CONTRIBUTING.md]: (repeat as necessary)

**Comment by Klaas-** [comment]
Hi,
I think you need to change some more to also put postgresql10 into the awx containers. #2853

the changes to installer/roles/image_build/templates/Dockerfile.j2 are still needed, right? that will fix #2392

**Review by ryanpetrello — APPROVED** [review]

**Comment by ryanpetrello** [comment]
recheck

**Review by shanemcd — APPROVED** [review]

**Comment by shanemcd** [comment]
recheck

**Comment by Klaas-** [comment]
Are there any upgrade docs planned? There is https://github.com/sclorg/postgresql-container/tree/generated/10#upgrading-database-by-switching-to-newer-postgresql-image-version for the pg containers

**Pull request merged**